packages/ui/components/checkbox.tsx

```diff
@@ -5,31 +5,59 @@ import { Check } from "lucide-react";
 
 import { cn } from "../utils";
 
-const Checkbox = React.forwardRef<React.ElementRef<typeof CheckboxPrimitive.Root>, React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>>(
-  ({ className, ...props }, ref) => (
-    <CheckboxPrimitive.Root
-      ref={ref}
-      className={cn(
-        "border-primary ring-offset-background data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground focus-visible:ring-ring peer h-4 w-4 shrink-0 rounded-sm border focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
-        className
-      )}
-      {...props}
+interface CheckboxTouchTargetProps {
+  expandedTouchTarget?: boolean;
+}
+
+const Checkbox = React.forwardRef<
+  React.ElementRef<typeof CheckboxPrimitive.Root>,
+  React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root> & CheckboxTouchTargetProps
+>(({ className, onClick, expandedTouchTarget = false, ...props }, ref) => {
+  const checkboxRef = React.useRef<HTMLButtonElement>(null);
+  const wrapperRef = React.useRef<HTMLDivElement>(null);
+
+  React.useImperativeHandle(ref, () => checkboxRef.current!);
+
+  const handleWrapperClick = (e: React.MouseEvent<HTMLDivElement>) => {
```

Comment:
could you call is smth like
```suggestion
  const triggerCheckboxClick = (e: React.MouseEvent<HTMLDivElement>) => {
```

just to make the code a bit more readable